components/admin/lessons/AdminModuleInputs.tsx

```diff
@@ -0,0 +1,185 @@
+  title,
+  lessonId,
+  onAddModule,
+  module,
+  refetch
+}: Props) => {
+  const [formOptions, setFormOptions] = useState<any>(initValues())
+  const [name, content] = formOptions
+
+  useEffect(
+    () => setFormOptions(initValues(module?.name, module?.content)),
+    [module]
+  )
+
+  const [moduleMutation, { data, error, loading }] = module
+    ? useUpdateModuleMutation({
+        variables: {
+          id: module.id,
+          content: content.value,
+          lessonId,
+          name: name.value
+        }
+      })
+    : useAddModuleMutation({
+        variables: {
+          content: content.value,
+          lessonId,
+          name: name.value
+        }
+      })
+
+  const [errorMsg, setErrorMsg] = useState(error?.message || '')
+
+  const handleChange = async (value: string, propertyIndex: number) => {
+    await formChange(value, propertyIndex, formOptions, setFormOptions)
+  }
+
+  const onSubmit = async () => {
+    try {
+      // Empty strings aren't handled in the resolver
+      if (!content.value || !name.value) {
+        return setErrorMsg(Error.InvalidData)
+      }
+
+      setErrorMsg('')
+      const newModule = await moduleMutation()
+      refetch()
+
+      if (onAddModule) {
+        const updateModuleData = get(newModule, 'data.updateModule')
+        const addModuleData = get(newModule, 'data.addModule')
+
+        onAddModule(
+          (updateModuleData && {
+            ...updateModuleData,
+            lesson: { id: lessonId }
+          }) ||
+            (addModuleData && {
+              ...addModuleData,
+              lesson: { id: lessonId }
+            }) ||
+            null,
+          null
+        )
+      }
+    } catch (err) {
+      if (onAddModule) {
+        onAddModule(null, {
+          content: content.value,
+          name: name.value
+        })
+      }
+
+      setErrorMsg((err as ApolloError).message)
+    }
+  }
+
+  const QueryStateMessage = () => {
+    if (loading) {
+      return (
+        <div className={styles.loading}>
+          <Spinner animation="grow" size="sm" />
+          <span>Adding the module...</span>
+        </div>
+      )
+    }
+
+    if (errorMsg) {
+      return (
+        <div className={styles.error}>
+          <AlertFillIcon />
+          <span>Failed to add the module: {errorMsg}</span>
+        </div>
+      )
+    }
+
+    if (data) {
+      const updateModule = get(data, 'updateModule')
+      const addModule = get(data, 'addModule')
+
+      return (
+        <div className={styles.success}>
+          <CheckCircleIcon />
+          <span>
+            {updateModule ? 'Updated' : 'Added'} the module{' '}
+            <strong>{addModule?.name || updateModule?.name || ''}</strong>{' '}
+            successfully!
+          </span>
+        </div>
+      )
+    }
+
+    return <></>
+  }
+
+  return (
+    <div className={styles.container}>
+      <QueryStateMessage />
+      <FormCard
+        title={name.value || title || 'Untitled'}
```

Comment:
`name.value` is used to dynamically fill the title as the admin is typing it in the input.